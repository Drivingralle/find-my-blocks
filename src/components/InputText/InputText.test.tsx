import React from "react";
import renderer from "react-test-renderer";
import { cleanup, fireEvent, render } from "@testing-library/react";
import { InputText } from "./";

afterEach(cleanup);

it("render correctly", () => {
  const tree = renderer
    .create(<InputText onChange={() => undefined} defaultValue="hello" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("calls the onchange function when typed in", () => {
  const onChange = jest.fn();
  const { getByPlaceholderText } = render(
    <InputText placeholder="Test" onChange={() => onChange()} />
  );
  const input = getByPlaceholderText("Test");
  fireEvent.change(input, {
    target: { value: "Hello" },
  });

  expect(onChange).toHaveBeenCalled();
});

it("does nothing if onChange is not set", () => {
  // eslint-disable-next-line no-unused-vars
  const onChange = jest.fn();
  // @ts-ignore
  const { getByPlaceholderText } = render(<InputText placeholder="Test" />);
  const input = getByPlaceholderText("Test");
  fireEvent.change(input, {
    target: { value: "Hello" },
  });

  expect(onChange).not.toHaveBeenCalled();
});
