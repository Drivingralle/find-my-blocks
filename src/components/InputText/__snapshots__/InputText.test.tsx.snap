// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<input
  className="input"
  onChange={[Function]}
  type="text"
  value="hello"
/>
`;
